lib/rubygems/errors.rb

```diff
@@ -170,12 +170,21 @@ def initialize(source, error)
     # An English description of the error.
 
     def wordy
-      "Unable to download data from #{@source.uri} - #{@error.message}"
+      redact_password("Unable to download data from #{@source.uri} - #{@error.message}", @source.uri)
```

Comment:
actually, this should just create a password-less URI -- the current `gsub` approach could end up hitting more than just the password if the password is found elsewhere in the error message